contracts/ERC721A.sol

```diff
@@ -316,20 +316,26 @@ contract ERC721A is Context, ERC165, IERC721, IERC721Metadata, IERC721Enumerable
             addressData.balance + uint128(quantity),
             addressData.numberMinted + uint128(quantity)
         );
-        _ownerships[startTokenId] = TokenOwnership(to, uint64(block.timestamp));
 
-        uint256 updatedIndex = startTokenId;
+        uint256 batches = quantity / maxBatchSize;
+        if (quantity % maxBatchSize != 0) {
+            batches += 1;
+        }
+
+        // Recording ownerships in increments of maxBatchSize.
+        for (uint256 i = 0; i < batches; i++) {
+            _ownerships[startTokenId + maxBatchSize * i] = TokenOwnership(to, uint64(block.timestamp));
+        }
```

Comment:
First i = 0, so we always start with `startTokenId` because `maxBatchSize * i` will zero itself out.